crates/flux-refineck/src/fixpoint_encoding.rs

```diff
@@ -439,11 +443,20 @@ where
         })
     }
 
-    pub fn pred_to_fixpoint(&mut self, pred: &rty::Expr) -> (Bindings, PredSpans) {
-        let mut bindings = vec![];
-        let mut preds = vec![];
-        self.pred_to_fixpoint_internal(pred, &mut bindings, &mut preds);
-        (bindings, preds)
+    pub fn pred_to_fixpoint(&mut self, pred: &rty::Pred) -> (Bindings, PredSpans) {
+        match pred {
+            rty::Pred::Expr(expr) => {
+                let mut bindings = vec![];
+                let mut preds = vec![];
+                self.pred_to_fixpoint_internal(expr, &mut bindings, &mut preds);
+                (bindings, preds)
+            }
+            rty::Pred::Alias(alias_pred, args) => {
+                let func = self.alias_pred_func(alias_pred, args.len());
+                let alias_pred_app = rty::Expr::app(func, args.clone(), None);
+                self.pred_to_fixpoint(&rty::Pred::Expr(alias_pred_app))
```

Comment:
This is representing the associated predicate application as `rty::Expr::app` just to immediately encode it. We can go directly to fixpoint. I'd have something like this

```suggestion
                let func = self.register_const_for_alias_pred(alias_pred);
                let args = self.exprs_to_fixpoint(args);
                let pred = fixpoint::Expr::App(func, args);
                (vec![], vec![(pred, None)])
```

The trick is in `register_const_for_alias_pred` which has the following signature

```rust
fn register_const_for_alias_pred(&mut self, alias_pred: &rty::AliasPred) -> fixpoint::Var
```

This function registers a `ConstInfo` for the `AliasPred` (if there's no already one) and returns a `GlobalVar` for it. Note that the `ConstInfo` generated for an `AliasPred` should not go into the `ConstMap` because we never look it up by name (the lookup up key is the `AliasPred`).